components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -483,32 +472,26 @@ private void addUserstoreProperties(UserStoreDTO userStoreDTO, UserStoreListResp
     }
 
     /**
-     * Construct the configured user store type's meta.
+     * Construct the user store type's meta.
      *
-     * @param userStoreDTOS the list of {@link UserStoreDTO}.
      * @param typeId        the type id of the user store.
      * @return MetaUserStoreType.
      */
-    private List<MetaUserStoreType> buildUserStoreMetaResponse(List<UserStoreDTO> userStoreDTOS, String typeId) {
-
-        List<MetaUserStoreType> metaUserStoreTypes = new ArrayList<>();
-        for (UserStoreDTO userStoreDTO : userStoreDTOS) {
-            Properties properties = UserStoreManagerRegistry.getUserStoreProperties(userStoreDTO.getClassName());
-            MetaUserStoreType metaUserStore = new MetaUserStoreType();
-            UserStorePropertiesRes userStorePropertiesRes = new UserStorePropertiesRes();
-            userStorePropertiesRes.mandatory(buildPropertiesRes(properties.getMandatoryProperties()));
-            userStorePropertiesRes.optional(buildPropertiesRes(properties.getOptionalProperties()));
-            userStorePropertiesRes.advanced(buildPropertiesRes(properties.getAdvancedProperties()));
-            metaUserStore.setProperties(userStorePropertiesRes);
-            metaUserStore.setTypeId(typeId);
-            metaUserStore.setName(userStoreDTO.getDomainId());
-            metaUserStore.setDescription(userStoreDTO.getDescription());
-            metaUserStore.setTypeName(base64URLDecodeId(typeId));
-            metaUserStore.setClassName(getUserStoreType(base64URLDecodeId(typeId)));
-            metaUserStoreTypes.add(metaUserStore);
-
-        }
-        return metaUserStoreTypes;
+    private MetaUserStoreType buildUserStoreMetaResponse(String typeId) {
+
+        Properties properties = UserStoreManagerRegistry.getUserStoreProperties(
+                getUserStoreType(base64URLDecodeId(typeId)));
+        MetaUserStoreType metaUserStore = new MetaUserStoreType();
+        UserStorePropertiesRes userStorePropertiesRes = new UserStorePropertiesRes();
+        userStorePropertiesRes.mandatory(buildPropertiesRes(properties.getMandatoryProperties()));
```

Comment:
check whether properties is null before calling its methods

Reply:
addressed in https://github.com/wso2/identity-api-server/pull/122/commits/22314815863e0cfd7d96fc976fd0084f6a288512